server/app/auth/ProfileFactory.java

```diff
@@ -18,15 +22,18 @@ public class ProfileFactory {
   private DatabaseExecutionContext dbContext;
   private HttpExecutionContext httpContext;
   private Provider<VersionRepository> versionRepositoryProvider;
+  private SyncCacheApi syncCacheApi;
 
   @Inject
   public ProfileFactory(
       DatabaseExecutionContext dbContext,
       HttpExecutionContext httpContext,
-      Provider<VersionRepository> versionRepositoryProvider) {
+      Provider<VersionRepository> versionRepositoryProvider,
+      @NamedCache("api-keys") SyncCacheApi syncCacheApi) {
```

Comment:
Optional.empty() comment here too as it's part of the api we're adding to the built in cache.